service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerDateOfIssueMidEventTest.java

```diff
@@ -0,0 +1,208 @@
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.fpl.enums.DirectionAssignee.ALL_PARTIES;
+import static uk.gov.hmcts.reform.fpl.enums.DirectionAssignee.CAFCASS;
+import static uk.gov.hmcts.reform.fpl.enums.DirectionAssignee.COURT;
+import static uk.gov.hmcts.reform.fpl.enums.DirectionAssignee.LOCAL_AUTHORITY;
+import static uk.gov.hmcts.reform.fpl.enums.DirectionAssignee.OTHERS;
+import static uk.gov.hmcts.reform.fpl.enums.DirectionAssignee.PARENTS_AND_RESPONDENTS;
+import static uk.gov.hmcts.reform.fpl.enums.JudgeOrMagistrateTitle.HIS_HONOUR_JUDGE;
+import static uk.gov.hmcts.reform.fpl.utils.CaseDataGeneratorHelper.createHearingBooking;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.wrapElements;
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(DraftOrdersController.class)
+@OverrideAutoConfiguration(enabled = true)
+class DraftOrdersControllerDateOfIssueMidEventTest extends AbstractControllerTest {
+
+    DraftOrdersControllerDateOfIssueMidEventTest() {
+        super("draft-standard-directions");
+    }
+
+    @Nested
+    class DateOfIssueValidation {
+
+        @Test
+        void shouldReturnErrorsWhenTheDateOfIssueIsInFuture() {
+            CaseDetails caseDetails = caseDetails(now().plusDays(1));
+            AboutToStartOrSubmitCallbackResponse response = postMidEvent(caseDetails, "date-of-issue");
+            assertThat(response.getErrors()).containsOnlyOnce("Date of issue cannot be in the future");
+        }
+
+        @Test
+        void shouldNotReturnErrorsWhenDateOfIssueIsToday() {
+            CaseDetails caseDetails = caseDetails(now());
+            AboutToStartOrSubmitCallbackResponse response = postMidEvent(caseDetails, "date-of-issue");
+            assertThat(response.getErrors()).isNull();
+        }
+
+        @Test
+        void shouldNotReturnErrorsWhenDateOfIssueIsInPast() {
+            CaseDetails caseDetails = caseDetails(now().minusDays(1));
+            AboutToStartOrSubmitCallbackResponse response = postMidEvent(caseDetails, "date-of-issue");
+            assertThat(response.getErrors()).isNull();
+        }
+
+        private CaseDetails caseDetails(LocalDateTime localDateTime) {
+            return CaseDetails.builder().data(Map.of("dateOfIssue", localDateTime.toLocalDate())).build();
+        }
+    }
+
+    @Test
+    void shouldPopulateCorrectHearingDate() {
+        LocalDateTime hearingDate = LocalDateTime.of(2020, 1, 1, 0, 0, 0);
+
+        CaseDetails caseDetails = CaseDetails.builder()
+            .data(Map.of("hearingDetails", wrapElements(createHearingBooking(hearingDate, hearingDate.plusDays(1)))))
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(caseDetails, "date-of-issue");
+
+        Stream.of(DirectionAssignee.values()).forEach(assignee ->
+            AssertionsForClassTypes.assertThat(callbackResponse.getData().get(assignee.toHearingDateField()))
```

Comment:
`AssertionsForClassTypes` we don't need to use that type of assertion, it is for java 8 when it couldn't resolve the proper `assertThat` to call. [See here](https://joel-costigliola.github.io/assertj/core-8/api/org/assertj/core/api/AssertionsForClassTypes.html)

Reply:
yeah weird I thought I had removed all of this, good spot